### PR TITLE
Fixed client runtime

### PR DESCRIPTION
**Pull request by barambani**
This should reestablish correctness of the client and the CLI after the upgrade to the latest fs2. I will address possible performance regressions and the automation to build the CLI in follow-up PRs.

### CODE REVIEW

File: cli/src/main/scala/laserdisc/cli/CLI.scala [hunk 1]
```diff
@@ -60,90 +61,91 @@ object CLI extends IOApp.WithContext { self =>
...
+          }
 
-            val protocol: Pipe[IO, String, Protocol] = {
-              def mkProtocol(code: String): IO[Maybe[Protocol]] =
-                IO.delay {
-                  tb.eval {
-                      tb.parse {
-                        s"""import laserdisc._
-                           |import laserdisc.auto._
-                           |import laserdisc.all._
```
Comment (sirocchj) [inline cli/src/main/scala/laserdisc/cli/CLI.scala]: i think we need this for now (see [here](https://github.com/laserdisc-io/laserdisc/blob/master/core/src/main/scala/laserdisc/syntax.scala#L17)) or we lose the ability to just invoke all protocols w/o prefix in the CLI

File: fs2/src/main/scala/laserdisc/fs2/RedisChannel.scala [hunk 2]
```diff
@@ -4,6 +4,7 @@ package fs2
 import java.net.InetSocketAddress
 
 import _root_.fs2.{Chunk, Pull}
```
Comment (sirocchj) [inline fs2/src/main/scala/laserdisc/fs2/RedisChannel.scala]:
```suggestion
```

File: project/plugins.sbt [hunk 3]
```diff
@@ -7,3 +7,4 @@ addSbtPlugin("org.scala-js"       % "sbt-scalajs"              % "0.6.29")
 addSbtPlugin("pl.project13.scala" % "sbt-jmh"                  % "0.3.7")
 addSbtPlugin("org.scoverage"      % "sbt-scoverage"            % "1.6.0")
 addSbtPlugin("net.virtual-void"   % "sbt-dependency-graph"     % "0.10.0-RC1")
+addSbtPlugin("com.eed3si9n"       % "sbt-assembly"             % "0.14.10")
```
Comment (sirocchj) [inline project/plugins.sbt]: do we need this?

File: fs2/src/main/scala/laserdisc/fs2/RedisClient.scala [hunk 4]
```diff
@@ -155,79 +177,102 @@ object RedisClient {
                     case Some(failedAddress) if address === failedAddress =>
                       // this indicates that cluster sill thinks the address is same as the one that failed us, for that reason
                       // we have to suspend execution for while and retry in FiniteDuration
-                      logger.warnS(s"New server is same like the old one ($address): currently unavailable") >>
+                      LogWriter.warnS(s"New server is same like the old one ($address): currently unavailable") >>
                         serverUnavailable.flatMap(address => runner(Stream.emit(Some(address))))
 
-                    case _ =>
+                    case None =>
```
Comment (sirocchj) [inline fs2/src/main/scala/laserdisc/fs2/RedisClient.scala]: are we not missing a case now?

File: cli/src/main/scala/laserdisc/cli/CLI.scala (hunk 1 above)
Comment (barambani) [reply cli/src/main/scala/laserdisc/cli/CLI.scala]: I think the protocols in `all` are already imported by `laserdisc._`. If I add it and run
```
localhost:6379> strings.set("a", 23)
```
I get
```
reflective compilation has failed:

reference to strings is ambiguous;
it is imported twice in the same scope by
import laserdisc.all._
and import laserdisc._
localhost:6379>
```
The one that doesn't seem to be found is `BListP` though.

File: project/plugins.sbt (hunk 3 above)
Comment (barambani) [reply project/plugins.sbt]: I need it to produce the the CLI assembly. It doesn't work in `sbt` so I have to run it trough the Jar. That's still a problem, I couldn't find a way to make it progress in an sbt run. It hangs and needs to be killed. If started with
```
java -jar <the jar> localhost 6379
```
works fine.

File: fs2/src/main/scala/laserdisc/fs2/RedisClient.scala (hunk 4 above)
Comment (barambani) [reply fs2/src/main/scala/laserdisc/fs2/RedisClient.scala]: Right, fixing it.

File: cli/src/main/scala/laserdisc/cli/CLI.scala (hunk 1 above)
Comment (sirocchj) [reply cli/src/main/scala/laserdisc/cli/CLI.scala]: Yeah, in theory it should be just `set(“a”, 23)`

`strings` is another object that extends `StringP` (Like `all`, which extends all protocols)

File: project/plugins.sbt (hunk 3 above)
Comment (sirocchj) [reply project/plugins.sbt]: And yes, would love to understand why it hangs there if used within sbt

**Comment by sirocchj** [comment]
:shipit:

**Comment by barambani** [comment]
I'm spending a bit of time trying find out why CLI hangs or, if I fork the jvm on run, doesn't stop waiting for input but successfully finishes.

**Comment by barambani** [comment]
I pushed the missing import. I have another strategy in mind for the CLI, but it deserves a separate PR. I would merge this as it is then.

**Pull request merged**